PhysicsTools/PyTorch/test/testBaseCUDA.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Base class for tests.
+ *
+ */
+
+#ifndef PHYSICSTOOLS_PYTORCH_TEST_TESTBASECUDA_H
+#define PHYSICSTOOLS_PYTORCH_TEST_TESTBASECUDA_H
+
+#include <boost/filesystem.hpp>
+#include <filesystem>
+#include <cppunit/extensions/HelperMacros.h>
+#include <stdexcept>
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSetReader/interface/ParameterSetReader.h"
+#include "FWCore/PluginManager/interface/PluginManager.h"
+#include "FWCore/PluginManager/interface/standard.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/ServiceRegistry/interface/ServiceRegistry.h"
+#include "FWCore/ServiceRegistry/interface/ServiceToken.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "FWCore/Utilities/interface/ResourceInformation.h"
+#include "HeterogeneousCore/CUDAServices/interface/CUDAInterface.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/requireDevices.h"
```

Comment:
This `#include` seems to be unused.

Reply:
it is used to test the presence of devices `if (!cms::cudatest::testDevices())`

Reply:
Thanks. I see the call to `cms::cudatest::testDevices()` is in `testTorchSimpleDnnCUDA.cc`, so this `#include` should be moved there.

On the other hand, that check is not really needed, because (nowadays) `scram b runtests` will skip the test if the node has no GPU. And in the case of `USER_UNIT_TESTS=cuda scram b runtests` we want the test program to fail. Therefore, I'd suggest to either remove the call to `cms::cudatest::testDevices()`, or if you want more clear error message in case of devices not being available, use the `not testDevices()` to print the clearer error message and fail the test program.